toolbox/release_note_collector/main.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/github"
+	u "istio.io/test-infra/toolbox/util"
+)
+
+const (
+	releaseNoteSuffix = ".releasenote"
+)
+
+var (
+	org       = flag.String("user", "istio", "Github owner or org")
+	repos     = flag.String("repos", "", "Github repos, separate using \",\"")
+	label     = flag.String("label", "release-note", "Release-note label")
+	sort      = flag.String("sort", "create", "The sort field. Can be comments, created, or updated.")
+	order     = flag.String("order", "desc", "The sort order if sort parameter is provided. One of asc or desc.")
+	output    = flag.String("output", "./", "Path to output file")
+	previousRelease = flag.String("previous_release", "", "Previous release")
+	currentRelease   = flag.String("current_release", "", "Current release")
+	PRLink = flag.Bool("pr_link", false, "Weather a link of the PR is added at the end of each release note")
+	gh *u.GithubClient
+)
+
+func main() {
+	flag.Parse()
+	if *previousRelease == "" {
+		log.Printf("Error: You need to specfy a previous release")
+		os.Exit(1)
+	}
+	gh = u.NewGithubClientNoAuth(*org)
+
+	repoList := strings.Split(*repos, ",")
+	for _, repo := range repoList {
+		log.Printf("Start fetching release note from %s", repo)
+		queries, err := createQueryString(repo)
+		if err != nil {
+			log.Printf("Failed to create query string for %s", repo)
+			continue
+		}
+		log.Printf("Query: %v", queries)
+
+		issuesResult, err := gh.SearchIssues(queries, "", *sort, *order)
+		if err != nil {
+			log.Printf("Failed to fetch PR with release note for %s: %s", repo, err)
+			continue
+		}
+		if err = fetchRelaseNoteFromRepo(repo, issuesResult); err != nil {
+			log.Printf("Failed to get release note for %s: %s", repo, err)
+			continue
+		}
+	}
+
+}
+
+func fetchRelaseNoteFromRepo(repo string, issuesResult *github.IssuesSearchResult) error {
+	fileName := filepath.Join(*output, repo+releaseNoteSuffix)
+	f, err := os.OpenFile(fileName, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0600)
```

Comment:
Could you move the file creation outside of this function and just pass an io.Writer instead ? That way we could use one file for all repos.